ArkEcosystem/explorer
# fix: handle case when cryptocompare is down and cache

https://app.clickup.com/t/m59ftw

Refactor the way we fetch the data from cryptocompare to:
- Retrieve the information in a scheduled job and not at the moment
- Retrieve price, price change, and marketcap for all currencies on a single request
- Retrieve the 24hours history for every currency once every hour and with a 1-minute difference between each one, in case it fails to get the data it will try again after 10 minutes

To test:
1. run `php artisan explorer:cache-currencies-data` to cache the price-related data
2. run `php artisan explorer:cache-currencies-history --no-delay` to cache the price histories

Everything should still work and look as always.

To test cryptocompare down

1. Block cryptocompare by adding `127.0.0.1       min-api.cryptocompare.com` to you `/etc/hosts` file if you use Linux or Mac (we can discuss other alternatives on slack) and run `php artisan explorer:cache-currencies-data` and `php artisan explorer:cache-currencies-history --no-delay` again. The request will fail and if you keep open the home page you should see "not available" state, 

![Screen Shot 2021-06-08 at 15 29 52](https://user-images.githubusercontent.com/17262776/121262843-615d4580-c87a-11eb-9486-519da7b3fae7.png)


## Checklist
